Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -0,0 +1,314 @@
+﻿using Kudu.Contracts.Tracing;
+using Kudu.Core.Infrastructure;
+using Kudu.Core.Tracing;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Kudu.Services.Performance
+{
+    internal static class ProfileManager
```

Comment:
@mathewc not sure what you mean here. ProcessController is added in NinjectServices and ProfileManager is just static methods being called from ProcessController which require no instance objects.